File: CMR-EC/P1/07-11-2024/IsPalindrome.java
public class IsPalindrome {
    public static void main(String[] args) {
        String str = "fclsalayalam";
        boolean ans = isPalindrome(str);
            System.out.println(ans);
        }
        
        private static boolean isPalindrome(String str) {
           int n = str.length();
           if(n<=1) return true;
           if(str.charAt(0) != str.charAt(n-1)) return false;
           String subStr = str.substring(1, n-1);
           return isPalindrome(subStr);
        }
}

File: Python/IsPalindrome.py
def isPalindrome(str):
    n = len(str)
    if n <= 1: return True
    if str[0] != str[-1]: return False
    subStr = str[1:n-1]
    return isPalindrome(subStr)
def main():
    str = ""
    ans = isPalindrome(str)
    print(ans)
main()